lib/yarp.rb

```diff
@@ -102,6 +102,11 @@ def ==(other)
         other.end_offset == end_offset
     end
 
+    # Returns a new location that is the union of this location and the other.
+    def to(other)
```

Comment:
`#to` is a slightly confusing name for me here, what about `#union`?

Reply:
In TruffleRuby we call this `extend`: https://github.com/oracle/truffleruby/blob/4468b51ad29f7b4b197024e2baf9b0137eb75eed/src/main/java/org/truffleruby/parser/ast/ParseNode.java#L91-L98

`union(2..3, 6..7)` might be expected to yield 2..3 + 6..7 and not 2..7.

Reply:
@jemmaissroff I updated it to `join`, which mirrors what the parser gem has. I didn't like `extend` only because it's already a Kernel method.

Reply:
Given the current unidirectional semantics, a good name might be `until`: `loc.until(other_loc)`.
`join`/`union` sounds like they would be commutative, but the impl is not.